lisa/testsuite.py

```diff
@@ -84,6 +84,35 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
     def name(self) -> str:
         return self.runtime_data.metadata.name
 
+    def handle_exception(
+        self, exception: Exception, log: Logger, phase: str = ""
+    ) -> None:
+        if phase:
+            phase = f"{phase} "
+        if isinstance(exception, SkippedException):
+            ...
+            log.info(f"case skipped: {exception}")
+            log.debug("case skipped", exc_info=exception)
+            # case is skipped dynamically
+            self.set_status(TestStatus.SKIPPED, f"{phase}skipped: {exception}")
+        elif isinstance(exception, NotRunException):
+            log.info(f"case keep NOTRUN: {exception}")
+            log.debug("case NOTRUN", exc_info=exception)
+            # case is not run dynamically.
+            self.set_status(TestStatus.NOTRUN, f"{phase}notrun: {exception}")
+        elif isinstance(PassedException, NotRunException):
```

Comment:
Do you mean exception instead of PassedException?

Reply:
fix in next commits, will move it here.